### PR TITLE
ADX-625 Add BulkFileUploaderComponent

### PR DESCRIPTION
**Pull request by Manoj-nathwani**
- Jira ticket https://fjelltopp.atlassian.net/browse/ADX-625
- Video of UI https://youtu.be/zo1UgaCsc5Y
- Related https://github.com/fjelltopp/adx_develop/pull/56

TODO:
- [ ] ~Drag+drop to work on entire page (not just the element area)~
- [x] Progress bars showing upload progress (not absolutely required so I stopped after 2hrs of trying to get it working)
- [x] Remove `ckanext-file_uploader_ui` (see https://github.com/fjelltopp/adx_develop/pull/56)
- [x] A lot of small improvements were made to how we integrate react into ckan. It'll require a little refactoring but these should be copied over to the `FileInputComponent`. This is a 1hr task max.
- [x] TESTS! (should be a half day job as this component is very similar to `FileInputComponent`)
- [x] Update translations `.po` file

### CODE REVIEW

**Comment by Manoj-nathwani** [comment]
@tomeksabala This is ready for it's first PR review. I've got it to a state where it perfectly works however as it was a POC and we wanted this done asap i've taken the shortcuts mentioned above. Give it a test!

**Comment by Manoj-nathwani** [comment]
image

Error handling looks like this ^

File: ckanext/unaids/react/components/BulkFileUploaderComponent/src/FileUploader.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { useDropzone } from 'react-dropzone'
+
+export default function FileUploader({ setPendingFiles }) {
+    const maxResourceSize = 5; // TODO: set up correctly
```
Comment (Manoj-nathwani) [inline ckanext/unaids/react/components/BulkFileUploaderComponent/src/FileUploader.js:5]: don't forget about this!

File: ckanext/unaids/helpers.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # encoding: utf-8
 from ckan.lib.helpers import url_for_static_or_external, check_access
+from ckanext.scheming.helpers import scheming_get_dataset_schema
```
Comment (Manoj-nathwani) [inline ckanext/unaids/helpers.py:3]: No idea what this is but I seem to have added it 🤔

File: ckanext/unaids/theme/templates/package/new_resource_not_draft.html [hunk 3]
```diff
@@ -24,15 +24,4 @@
...
-  {%- set schema = h.scheming_get_dataset_schema(dataset_type) -%}
-  {%- set resource_schema = schema.get("resource_schemas",{}).get(resource_type, {}) -%}
-  {% if not resource_schema.get('bulk_upload', False) %}
-  <section class="module module-narrow module-shallow">
-      <h2 class="module-heading"><i class="fa fa-upload"></i> {{ _('Bulk Upload') }}</h2>
-      <div class="module-content"><p>{{ _('Bulk upload is not possible for this resource.') }}</p></div>
-  </section>
-  {% else %}
-    {% include 'ckanext-file_uploader_ui/upload.html' %}
-  {% endif %}
```
Comment (tomeksabala) [inline ckanext/unaids/theme/templates/package/new_resource_not_draft.html:37]: Please keep this buld upload toggle in `ckanext/unaids/theme/templates/package/read_base.html`

File: ckanext/unaids/react/components/BulkFileUploaderComponent/src/App.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from 'react';
+import { Promise } from 'bluebird';
+import axios from 'axios';
+import FileUploader from './FileUploader';
+import { Client } from 'giftless-client';
+import ProgressBar from './ProgressBar';
+
+export default function App({ lfsServer, maxResourceSize, orgId, datasetId, defaultFields }) {
```
Comment (tomeksabala) [inline ckanext/unaids/react/components/BulkFileUploaderComponent/src/App.js:8]: Try to organize your react components into separate files for better readability.

**Review by tomeksabala — CHANGES_REQUESTED** [review]

**Pull request merged**